packages/flutter_test/lib/src/test_text_input.dart

```diff
@@ -36,6 +38,9 @@ class TestTextInput {
   /// first be requested, e.g. using [WidgetTester.showKeyboard].
   final VoidCallback onCleared;
 
+  /// The messenger which sends the bytes for this channel, not null.
+  BinaryMessenger binaryMessenger;
```

Comment:
should be final

Reply:
or just use the one from the services binding directly.

in fact here and various other places, i think it's better to do `ServicesBinding.instance.defaultBinaryMessenger` directly than to use a local copy. Alternatively, use a private `_binaryMessenger` getter that just returns `ServicesBinding.instance.defaultBinaryMessenger`. This avoids the local copy.

Reply:
I like the latter approach. It's generally a good practice to keep external dependencies obvious. Also, easier for injecting mocks for testing.